server/src/test/java/org/elasticsearch/action/bulk/BulkProcessorRetryIT.java

```diff
@@ -114,9 +114,9 @@ public void afterBulk(long executionId, BulkRequest request, Throwable failure)
                 for (BulkItemResponse bulkItemResponse : bulkResponse.getItems()) {
                     if (bulkItemResponse.isFailed()) {
                         BulkItemResponse.Failure failure = bulkItemResponse.getFailure();
-                        Throwable rootCause = ExceptionsHelper.unwrapCause(failure.getCause());
-                        if (rootCause instanceof EsRejectedExecutionException) {
-                            if (rejectedExecutionExpected == false) {
+                        if (failure.getStatus() == RestStatus.TOO_MANY_REQUESTS) {
+                            if (!rejectedExecutionExpected) {
```

Comment:
can you leave the previous `rejectedExecutionExpected == false` please? we prefer this one for readability.

Reply:
Could you address this please?